reana_server/rest/workflows.py

```diff
@@ -1214,7 +1214,7 @@ def download_file(workflow_id_or_name, file_name):  # noqa
           stream=True)
         return Response(
             stream_with_context(req.iter_content(chunk_size=1024)),
-            content_type=req.headers['Content-Type']), 200
+            content_type=req.headers['Content-Type']), req.status_code
```

Comment:
Works well :+1: 

Can we replace the user-visible message that currently says:
```
$ reana-client download wrong.txt -w testyad
File wrong.txt could not be downloaded: Expected status code 200 but replied with 404
```

This message is too much developer-parlance oriented.

Can we replace it with something more user oriented like:

```
$ reana-client download wrong.txt -w testyad
File wrong.txt could not be downloaded.  Error 404: File not found.
```

(can be done in the client after merging this PR)

Reply:
Fixed on [client PR](https://github.com/reanahub/reana-client/pull/405)